File: src/main/java/graph/core/Vertex.java
package graph.core;

import java.awt.Point;

public class Vertex {
	public static final int NO_OBJECT = -1;
	
	private int id;
	private int objectId;
	private Point pos;
	
	public Vertex(int id) {
		this.id       = id;
		this.objectId = NO_OBJECT;
	}
	
	public int getId() {
		return id;
	}
	
	public void setObjectId(int id) {
		objectId = id;
	}
	
	public int getObjectId() {
		return objectId;
	}

	public String toString() {
		return this.id + "";
	}

	public void setNodeLocation(int x, int y) {
		pos = new Point(x, y);
	}
	
	public Point getNodeLocation() {
		return pos;
	}
	
	public boolean containObject() {
		return objectId != Vertex.NO_OBJECT;
	}
}

File: src/main/scala/graph/Vertex.scala
package graph

import java.awt.Point
import Vertex._

/**
 * Created: 31.12.13, 13:43
 * @author fliebhart
 */

object Vertex {
  val NO_OBJECT = -1
}

class Vertex(var id: Int) {

  var objectId: Int = NO_OBJECT
  private var pos: Point = _

  override def toString: String = this.id + ""

  def setNodeLocation(x: Int, y: Int) {
    pos = new Point(x, y)
  }

  def getNodeLocation: Point = pos

  def containsObject: Boolean = objectId != Vertex.NO_OBJECT
}
